src/App.tsx

```diff
@@ -8,63 +8,78 @@ import { TodoList } from './components/TodoList';
 import { TodoError } from './components/TodoError';
 import { TodoFooter } from './components/TodoFooter';
 import * as dataOperations from './api/todos';
-import { SelectedTodo } from './types/SelectedTodo';
+import { Filter } from './types/Filter';
+
+const filterTodos = (todos: Todo[], filter: Filter | ''): Todo[] => {
+  switch (filter) {
+    case Filter.all:
+      return todos;
+    case Filter.active:
+      return todos.filter((todo) => !todo.completed);
+    case Filter.completed:
+      return todos.filter((todo) => todo.completed);
+    default:
+      return todos;
+  }
+};
```

Comment:
let's simplify your fn a bit)

```typescript
const filterTodos = (todos: Todo[], filter: Filter | ''): Todo[] => {
  return todos.filter(({completed}) => {
    switch (filter) {
      case Filter.active:
        return !completed;
        
      case Filter.completed:
        return completed;
        
      default: 
        return true;
    }
  }) 
};
``` 

please implement it in next parts of this task